### DISCUSSION TITLE
Passing large csv files into statically-exported pages: pass in as `query`?

### DISCUSSION THREAD

**Discussion by erikdstock** [vercel/next.js#10762] [Help]
I'd like to pass all csvs in a `data/` directory into individual statically-exported pages, but i'm not clear on the best way to do it. In pseudocode terms i've been doing something like this:
```
exportPathMap() {
  const paths = {}
  dataFiles.forEach(file => {
    const csv = fs.readFileSync(file + ".csv", 'utf-8')
    paths["/results/" + file] = {page: "/results/[result]", query: { csv }}
  }
  
  return paths
}
```
Then in my `getInitialProps` i do more complicated logic on the csv string, parsing it into an object and so on (this is not easy to do from next.config.js due to the stripped-down node environment).

That said, this all feels weird. I'm coming from a gatsby background in part but wanted to try next since these csvs felt like they didn't call for a full graphql type system, but it feels like i'm doing something wrong to be assigning `csv` to a 'query' object. Am I missing something? Is there a much easier way to do things? Should i be waiting for the `getStaticProps` method?

(also sorry for asking back-to-back questions)

**Comment by timneutkens** [comment]
You'll probably want to use https://github.com/zeit/next.js/issues/9524 when it's out.

**Reply by erikdstock** [reply]
Yeah, i've got my eye on it (actually commented there a few days ago)- but I also want to understand the nextjs context + best practices until then.
